static/src/javascripts/projects/common/modules/onward/history.js

```diff
@@ -0,0 +1,477 @@
+                            const newAge = day[0] + updateBy;
+                            return newAge < summaryPeriodDays && newAge >= 0
+                                ? [newAge, day[1]]
+                                : false;
+                        })
+                        .filter(Boolean);
+
+                    return visits.length > 1 ||
+                        (visits.length === 1 &&
+                            visits[0][0] < forgetUniquesAfter)
+                        ? visits
+                        : [];
+                }
+
+                return [];
+            });
+
+            if (result.some(r => r.length)) {
+                summary.tags[tid] = [record[0]].concat(result);
+            } else {
+                delete summary.tags[tid];
+            }
+        });
+    }
+
+    return summary;
+};
+
+const tally = (
+    visits: Array<Array<number>>,
+    weight: number = 1,
+    minimum: number = 1
+): number => {
+    let totalVisits = 0;
+
+    const result = visits.reduce((t, day) => {
+        const dayOffset = day[0];
+        const dayVisits = day[1];
+
+        totalVisits += dayVisits;
+        return t + weight * (9 + dayVisits) * (summaryPeriodDays - dayOffset);
+    }, 0);
+
+    return totalVisits < minimum ? 0 : result;
+};
+
+const getPopular = (opts: ?Object): Array<Array<string>> => {
+    const tags = getSummary().tags;
+    let tids = Object.keys(tags);
+
+    const op = Object.assign(
+        {},
+        {
+            number: 100,
+            weights: {},
+            thresholds: {},
+        },
+        opts
+    );
+
+    tids = op.whitelist
+        ? tids.filter(tid => op.whitelist.indexOf(tid) > -1)
+        : tids;
+    tids = op.blacklist
+        ? tids.filter(tid => op.blacklist.indexOf(tid) === -1)
+        : tids;
+
+    return tids
+        .map(tid => {
+            const record = tags[tid];
+            const rank = buckets.reduce(
+                (r, bucket) =>
+                    r +
+                    tally(
+                        record[bucket.indexInRecord],
+                        op.weights[bucket.type],
+                        op.thresholds[bucket.type]
+                    ),
+                0
+            );
+
+            return {
+                idAndName: [tid, record[0]],
+                rank,
+            };
+        })
+        .filter(Boolean)
+        .sort((a, b) => a.rank - b.rank)
+        .slice(-op.number)
+        .map(tid => tid.idAndName)
+        .reverse();
+};
+
+const getContributors = (): Array<any> => {
+    const contibutors = [];
+    const tags = getSummary().tags;
+
+    Object.keys(tags).forEach(tagId => {
+        if (tagId.indexOf('profile/') === 0) {
+            contibutors.push(tags[tagId]);
+        }
+    });
+
+    return contibutors;
+};
+
+const collapsePath = (t: string): string => {
+    const isEditionalisedRx = new RegExp(
+        `^(${editions.join('|')})\/(${editionalised.join('|')})$`
```

Comment:
Yes this can go, I'm sure I'd read that creating a new RegExp instance meant I'd need to keep this escape, I wrote the test to confirm it was OK to remove it but never actually tried to remove it after I'd written the test! :laughing: